airbyte-cdk/bulk/core/base/src/testFixtures/kotlin/io/airbyte/cdk/command/CliRunner.kt

```diff
@@ -33,54 +40,73 @@ data object CliRunner {
         config: ConfigurationJsonObjectBase? = null,
         catalog: ConfiguredAirbyteCatalog? = null,
         state: List<AirbyteStateMessage>? = null,
-    ): BufferingOutputConsumer =
+    ): BufferingOutputConsumer {
+        val out = TestOutputStream()
         runConnector(op, config, catalog, state) { args: Array<String> ->
-            AirbyteSourceRunner(args)
+            AirbyteSourceRunner(args, out.beanDefinition)
         }
+        return out.results
+    }
 
     /** Same as [runSource] but for destinations. */
     fun runDestination(
         op: String,
         config: ConfigurationJsonObjectBase? = null,
         catalog: ConfiguredAirbyteCatalog? = null,
         state: List<AirbyteStateMessage>? = null,
-    ): BufferingOutputConsumer =
+        inputStream: InputStream,
+    ): BufferingOutputConsumer {
+        val inputBeanDefinition: RuntimeBeanDefinition<InputStream> =
+            RuntimeBeanDefinition.builder(InputStream::class.java) { inputStream }
+                .singleton(true)
+                .build()
+        val out = TestOutputStream()
         runConnector(op, config, catalog, state) { args: Array<String> ->
-            AirbyteDestinationRunner(args)
+            AirbyteDestinationRunner(args, inputBeanDefinition, out.beanDefinition)
+        }
+        return out.results
+    }
+
+    /** Same as the other [runDestination] but with simpler input of [AirbyteMessage] instances. */
+    fun runDestination(
+        op: String,
+        config: ConfigurationJsonObjectBase? = null,
+        catalog: ConfiguredAirbyteCatalog? = null,
+        state: List<AirbyteStateMessage>? = null,
+        vararg input: AirbyteMessage,
+    ): BufferingOutputConsumer {
+        val baos = ByteArrayOutputStream()
+        for (msg in input) {
+            Jsons.writeValue(baos, msg)
+            baos.write('\n'.code)
         }
+        val inputStream = ByteArrayInputStream(baos.toByteArray())
```

Comment:
nit: can we skip the baos?
```kotlin
ByteArrayInputStream(msg.map { Jsons.serialize(it) + '\n' }.toByteArray())
```

and if not, then I think this structure makes it more obvious that the baos is just a temp thing:
```kotlin
val bytes = ByteArrayOutputStream().also { baos ->
  for (msg in input) {
    Jsons.writeValue(baos, msg)
    baos.write('\n'.code)
  }
}.toByteArray()
```